src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -92,41 +129,97 @@ function useSortedTokens(tokenData: TokenData[] | null) {
 
   return useMemo(
     () =>
-      tokenData &&
-      tokenData.sort((token1, token2) => {
-        if (!tokenData) {
-          return 0
-        }
-        // fix delta/percent change property
-        if (!token1 || !token2 || !sortDirection || !sortCategory) {
-          return 0
-        }
-        let a: number | null | undefined
-        let b: number | null | undefined
-        switch (sortCategory) {
-          case Category.marketCap:
-            a = token1.marketCap?.value
-            b = token2.marketCap?.value
-            break
-          case Category.price:
-            a = token1.price?.value
-            b = token2.price?.value
-            break
-          case Category.volume:
-            a = token1.volume?.[timePeriod]?.value
-            b = token2.volume?.[timePeriod]?.value
-            break
-          case Category.percentChange:
-            a = token1.percentChange?.[timePeriod]?.value
-            b = token2.percentChange?.[timePeriod]?.value
-            break
-        }
-        return sortFn(a, b)
-      }),
-    [tokenData, sortDirection, sortCategory, sortFn, timePeriod]
+      nonDefaultSortingParamsExist
+        ? tokenData &&
+          tokenData.sort((token1, token2) => {
+            if (!tokenData) {
+              return 0
+            }
+            // fix delta/percent change property
+            if (!token1 || !token2 || !sortDirection || !sortCategory) {
+              return 0
+            }
+            let a: number | null | undefined
+            let b: number | null | undefined
+            switch (sortCategory) {
+              case Category.marketCap:
+                a = token1.marketCap?.value
+                b = token2.marketCap?.value
+                break
+              case Category.price:
+                a = token1.price?.value
+                b = token2.price?.value
+                break
+              case Category.volume:
+                a = token1.volume?.[timePeriod]?.value
+                b = token2.volume?.[timePeriod]?.value
+                break
+              case Category.percentChange:
+                a = token1.percentChange?.[timePeriod]?.value
+                b = token2.percentChange?.[timePeriod]?.value
+                break
+            }
+            return sortFn(a, b)
+          })
+        : tokenData,
+    [tokenData, sortDirection, sortCategory, sortFn, timePeriod, nonDefaultSortingParamsExist]
   )
 }
 
+function createFetchTokensPromise(
+  page: number,
+  pageSize: number,
+  startIndex: number,
+  setTopTokens: (topTokens: TokenData[]) => void,
+  setError: (error: any) => void,
+  topTokens: TokenData[]
+) {
+  fetchQuery<TopTokenQueryType>(environment, query, {
+    pageSize,
+    page,
+  })
+    .toPromise()
+    .then((data) => {
+      const newPageTopTokens: TokenData[] = !!data?.topTokenProjects
+        ? data.topTokenProjects.map((token) =>
```

Comment:
could you split this function out into a separate one to make it more readable? `formatTokenProjects` or something?